src/store/modules/accounts/hdWallet.js

```diff
@@ -30,16 +38,44 @@ export default {
       return aeSdk.api.getAccountByPubkey(address).then(() => true, () => false);
     },
     async discover({ rootGetters: { wallet }, dispatch }) {
-      const numberOfAccounts = await ProtocolAdapterFactory
-        .getAdapter(PROTOCOL_AETERNITY) // Discover only aeternity accounts for now
-        .discoverAccounts(wallet);
-
-      for (let i = 0; i < numberOfAccounts; i += 1) {
-        dispatch('create', {
-          isRestored: true,
-          protocol: PROTOCOL_AETERNITY,
-        });
-      }
+      const accountsToRecover = await Promise.all(PROTOCOLS.map(async (protocol) => {
```

Comment:
To be honest I don't like the fact that the logic of discovering the accounts was moved from the adapters. I know for AE and BTC the logic is the same and duplicating same code would make no sense. But maybe we could have an utility function for this purpose, which could be reused within the adapters? This would make the code more modular and make it easier to add new protocols that has different way of account discovery.